api/src/services/project-service.ts

```diff
@@ -431,7 +431,16 @@ export class ProjectService extends DBService {
     await this.updateProject(projectId, entities);
 
     try {
-      await this.platformService.submitProjectDwCMetadataToBioHub(projectId);
+      // Publish project metadata
+      const publishProjectPromise = this.platformService.submitProjectDwCMetadataToBioHub(projectId);
+
+      // Publish all survey metadata (which needs to be updated now that the project metadata has changed)
+      const publishSurveysPromise = (async () => {
```

Comment:
Could maybe be a bit more readible if we use `then(...)` instead of `(async () => ...)()`:

```typescript
const publishSurveysPromise = this.surveyService.getSurveyIdsByProjectId(projectId)
  .then((surveyIds) => {
    return surveyIds.map((surveyId) => this.platformService.submitSurveyDwCMetadataToBioHub(item.id));
  });
```

Reply:
Good call, was messing around with async stuff a bit there. But ya I agree its not as clean as it should be.

Reply:
Pushed up